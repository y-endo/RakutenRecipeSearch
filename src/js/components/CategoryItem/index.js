import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default class CategoryItem extends React.Component {
  render() {
    return (
      <li className="recipe-list__item">
        <Link to={`/category/${this.props.recipeCategoryData.categoryId}`} className="recipe-list__link">
          {this.props.recipeCategoryData.categoryName}
        </Link>
      </li>
    );
  }
}

CategoryItem.propTypes = {
  recipeCategoryData: PropTypes.object
};
