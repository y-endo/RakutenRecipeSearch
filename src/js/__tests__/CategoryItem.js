import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CategoryItem from '@/js/components/CategoryItem/index';

test('カテゴリーアイテムが正しく表示されるか', () => {
  const wrapper = shallow(<CategoryItem recipeCategoryData={{ categoryId: 10, categoryName: 'カテゴリー名' }} />);

  // カテゴリー名が正しい
  expect(wrapper.text()).toBe('カテゴリー名');

  // Linkのtoが正しい
  expect(wrapper.find('.recipe-list__link').props().to).toBe('/category/10');

  // スナップショット
  expect(toJson(wrapper)).toMatchSnapshot();
});
