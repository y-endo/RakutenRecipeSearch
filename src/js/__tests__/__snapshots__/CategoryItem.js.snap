// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`カテゴリーアイテムが正しく表示されるか 1`] = `
<li
  className="recipe-list__item"
>
  <Link
    className="recipe-list__link"
    to="/category/10"
  >
    カテゴリー名
  </Link>
</li>
`;
